Pairwise shifts

Dronte opened the pull request:
This commit introduces pairvise alignments for PWMs in DiffLogo.
We add localPwmAlignment function that bruteforces all the alignments and
counts divergence for each. Simply takes the best one. Also
it tests for reverse complement.

No plotting of the alignment is included in DiffLogo on this step.

There is  also a nice plotting fuction usefull for testing:
https://gist.github.com/Dronte/f39772d8565b04700fd80f88b45d7e49

mgledi commented on R/diffSeqLogo.R:
```diff
@@ -295,39 +294,191 @@ diffLogoTable = function (
+
+##' Finds best local alignment for PWMs.
+##'
+##' @title Aligh pwms
```
typo aligh = align

mgledi commented on R/diffSeqLogo.R:
```diff
@@ -295,39 +294,191 @@ diffLogoTable = function (
+
+findBestShiftForPwms = function(static_pwm, shifted_pwm, divergence,
+                                unaligned_penalty) {
+    best_divergence = 1/0 # +Inf
```
you can also write 'a=Inf'

Dronte committed:
Introdusing pairwise pwm alignments.

This commit introduces pairvise alignments for PWMs in DiffLogo.
We add localPwmAlignment function that bruteforces all the alignments and
counts divergence for each. Simply takes the best one. Also
it tests for reverse complement.

Dronte committed:
Documentation fixes and minor code formattig.

Dronte committed:
Typo fixes based on code review from @mgledi.

Dronte committed:
Fix a bug where 4 was hardcoded as alphabet size.

Dronte committed:
Fix syntax error.

Dronte committed:
Fixing bug with incorrect result of alignment.

Dronte committed:
Refactored alignment to make it consistant with further multiple alignment.
